components/scream/src/physics/p3/p3_constants.hpp

```diff
@@ -0,0 +1,63 @@
+#ifndef P3_CONSTANTS_HPP
+#define P3_CONSTANTS_HPP
+
+#include "share/scream_types.hpp"
+
+#include <vector>
+
+namespace scream {
+namespace p3 {
+
+/*
+ * Mathematical constants used by p3.
+ */
+
+template <typename Scalar>
+struct Constants
+{
+  static constexpr Scalar INV_RHOW = 1.e-3;
+  static constexpr Scalar RHOW     = 997.0;
+  static constexpr Scalar THRD     = 1.0/3.0;
+  static constexpr Scalar SXTH     = 1.0/6.0;
+  static constexpr Scalar PI       = 3.14159265;
+  static constexpr Scalar PIOV6    = PI*SXTH;
+  static constexpr Scalar CONS1    = PIOV6*RHOW;
+  static constexpr Scalar QSMALL   = 1.e-14;
+  static constexpr Scalar NSMALL   = 1.e-16;
+  static constexpr Scalar P0       = 100000.0;        // reference pressure, Pa
+  static constexpr Scalar RD       = 287.15;          // gas constant for dry air, J/kg/K
+  static constexpr Scalar RHOSUR   = P0/(RD*273.15);
+  static constexpr Scalar CP       = 1005.0;          // heat constant of air at constant pressure, J/kg
+  static constexpr Scalar INV_CP   = 1.0/CP;
+};
```

Comment:
I'd like to advocate for not using all-caps for these constants. Physical constants often have lower, upper, or mixed case by convention. For example, RD makes more sense as Rd or R_d. PI should be pi. These don't have to change all at once, but for the ones that are used in the Fortran P3, I think we should follow that convention. @bartgol, @AaronDonahue, thoughts?

More generally, since C++ has namespaces (via either `namespace` or, as in this case, structs and classes), we don't have to use special casing to indicate something is a global constant; the struct name `Constants` fulfills that role in this case.